api/src/services/config_service.rs

```diff
@@ -343,22 +446,148 @@ mod tests {
 
     #[test]
     fn should_return_application_companions_as_error_when_invalid_image_is_provided() {
-        let config_str = r#"
-            [[companions]]
+        let config = config_from_str!(
+            r#"
             [companions.openid]
             serviceName = 'openid'
             type = 'application'
             image = ''
             env = [ 'KEY=VALUE' ]
-        "#;
+            "#
+        );
 
-        let config = from_str::<Config>(config_str).unwrap();
-        let result = config.get_application_companion_configs();
+        let result = config.get_application_companion_configs("master");
 
         assert_eq!(result.is_err(), true);
         match result.err().unwrap() {
-            ConfigError::UnableToParseImage => assert!(true),
+            ConfigError::UnableToParseImage { image: _ } => assert!(true),
             _ => assert!(false, "unexpected error"),
         }
     }
+
+    #[test]
+    fn should_return_application_companions_with_specific_app_selector() {
+        let config = config_from_str!(
+            r#"
+            [companions.openid]
+            serviceName = 'openid'
+            type = 'application'
+            image = 'private.example.com/library/opendid:latest'
```

Comment:
🦅 👁 not a big deal, but `opendid`->`openid`?

Reply:
Nice catch. :+1: :eagle: